packages/elastic-handler/src/elasticHandler.ts

```diff
@@ -0,0 +1,98 @@
+import { Client, ClientOptions } from '@elastic/elasticsearch';
+
+import { LogEntry, LogLevel, TContextBase, LogHandler } from '@debugr/core';
+
+export interface ElasticHandlerOptions<
+  TContext extends TContextBase = { processId: string },
+  TGlobalContext extends Record<string, any> = {},
+> {
+  baseIndex?: string;
+  threshold: LogLevel | number;
+  indexCallback?: (entry: LogEntry<TContext, TGlobalContext>) => string;
+  errorCallback?: (error: Error) => void;
+  bodyParser?: (entry: LogEntry<TContext, TGlobalContext>) => Record<string, any>;
+  errorMsThreshold?: number;
+}
```

Comment:
zrušil bych `baseIndex` a `indexCallback` bych přejmenoval jen na `index`, kterej může bejt buď `string` ( = nejjednodušší usecase kdy chceš mít stejnej index na všechno) a nebo callback jako máš teď

`errorCallback` bych zrušil a místo toho z ElasticHandleru udělal event emitter kterej emituje `error` event (nebo to nějak sjednotil do globálního event dispatcheru

`bodyParser` zní jako že to něco _parsuje_, ale ono to spíš nějak formátuje / filtruje / mapuje, to bych taky zkusil nazvat nějak jinak

Reply:
nelibi se mi, ze v pripade erroru se emitne event... 
Za to rozhodne bych nebral moznost sem dat errorCallback... Kde jinde by se mel resit nez v handleru samotnem?

Reply:
well, `error` event je poměrně rozšířená konvence, a umožňuje na problém reagovat víc věcma zároveň - callback můžeš uvést jen jeden; a ten event by se emitoval přesně z toho místa, kde teď voláš `errorCallback`, takže by se to "řešilo" na tom samým místě.. další věc je defaultní chování když callback neuvedeš - teď to řešíš nějakým defaultním callbackem, ale za mě by klidně správný řešení bylo vyhodit v takovým případě výjimku, a to za tebe dokonce standardní event emitter myslím udělá sám pokud `error` event nemá handler..

Reply:
porad se mi nelibi ty eventy :D

Reply:
válka!!